conans/model/scm.py

```diff
@@ -86,7 +88,13 @@ def excluded_files(self):
     def checkout(self):
         output = ""
         if self._data.type == "git":
-            output += self.repo.clone(url=self._data.url)
+            try:
+                output += self.repo.clone(url=self._data.url, branch=self._data.revision, shallow=True)
```

Comment:
I'm bummed out that this feature was introduced as default True -- it broke dozens of our recipes :-(

https://github.com/conan-io/conan/issues/6068#issuecomment-555473862

Reply:
really sorry it caused issues for you :(
however, [SCM](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#scm) feature is currently marked as an experimental, and it's subject for breaking changes

Reply:
@michaelmaguire as I commented at #6068 (comment), if the shallow clone can't work in all cases we will revert the shallow to be opt-in. We don't want by default broken behaviors, obviously, so as @SSE4 said, we can break, but we will break in your favor if that is the case.